app/knights/knight.py

```diff
@@ -0,0 +1,24 @@
+class Knight:
+
+    def __init__(self, config: dict) -> None:
+
+        self.name = config["name"]
+        self.power = config["power"]
+        self.hp = config["hp"]
+        self.protection = 0
+
+        if len(config["armour"]) > 0:
+            for item in config["armour"]:
+                self.protection += item["protection"]

```

Comment:
The code assumes that 'armour' is always a list in the config. Consider checking if 'armour' exists and is a list before iterating over it to prevent potential errors.